lib/cogctl/actions/users/info.ex

```diff
@@ -47,4 +39,18 @@ defmodule Cogctl.Actions.Users.Info do
     Enum.map(permissions, fn(perm) ->
       "#{perm["namespace"]}:#{perm["name"]}" end)
   end
+
+  defp get_roles_from_groups(groups) do
+    Enum.reduce(groups, [], fn(group, roles) ->
+                roles ++ get_roles(group.roles)
+    end)
+    |> Enum.uniq
+  end
+
+  defp get_roles(roles) do
```

Comment:
Might be a good use for `get_in`?
```
all = fn(:get, data, next) ->
  Enum.flat_map(data, &next.(Map.delete(&1, :__struct__)))
end

roles = get_in(user.groups, [all, :roles])
|> Enum.uniq
```
Then in your table rendering logic, something like:
```
Enum.map(roles, fn(role) ->
  permissions = Enum.map_join(role.permissions, ",", &"#{&1["namespace"]}:#{&1["name"]}")
  [role.name, role.id, permissions]
end)
```